controllers/dataplane/openstackdataplanenodeset_controller.go

```diff
@@ -460,6 +464,7 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 
 func checkDeployment(ctx context.Context, helper *helper.Helper,
 	instance *dataplanev1.OpenStackDataPlaneNodeSet,
+	generationChanged bool,
```

Comment:
Since we already have the pointer to the instance in this function, it might be better to do the `generationChanged` computation within this function. Just avoids having to copy the variable between memory stacks when we call this function.

Reply:
Check reply to your other comment.